crates/prover/src/prove.rs

```diff
@@ -452,62 +459,172 @@ pub async fn prove(mut args: ProveArgs) -> anyhow::Result<bool> {
     // recursively combine expected proofs
     if !args.proving.skip_stitching() && result_pq.len() > 1 {
         // gather sorted proofs into vec
-        let results = result_pq
+        let mut results = result_pq
             .into_sorted_vec()
             .into_iter()
             .rev()
-            .map(|r| r.result.expect("Failed to get result"))
             .collect::<Vec<_>>();
 
-        // stitch contiguous proofs together
-        info!("Composing {} proofs together.", results.len());
-        // construct a proving instruction with no blocks to derive
-        let mut base_args = args.clone();
-        {
-            // set last block as starting point
-            base_args.kona.agreed_l2_output_root = base_args.kona.claimed_l2_output_root;
-            let l2_provider = l2_provider.as_ref().unwrap();
-            base_args.kona.agreed_l2_head_hash = await_tel!(
-                context,
-                tracer,
-                "l2_provider get_block_by_number claimed_l2_block_number",
-                retry_res_ctx_timeout!(
-                    args.timeouts.op_geth_timeout,
-                    l2_provider
-                        .get_block_by_number(BlockNumberOrTag::Number(
-                            base_args.kona.claimed_l2_block_number,
-                        ))
+        let l2_provider = l2_provider.as_ref().unwrap();
+        // collapse results vector by stitching its proofs
+        while results.len() > 1 {
+            let num_stitch_proofs = results.len().div_ceil(args.proving.max_proof_stitches);
+            let mut stitched_proof_receivers = Vec::with_capacity(num_stitch_proofs);
+            // compute a stitched proof for each subsequence of proofs
+            for stitched_proofs in results.chunks(args.proving.max_proof_stitches) {
```

Comment:
This stitching will run even when max_proof_stitches is 1. I don't think it breaks anything, but maybe just do the following to have it a bit cleaner:
```rust
pub fn skip_stitching(&self) -> bool {
    self.skip_derivation_proof || self.skip_await_proof || self.max_proof_stitches <= 1
}
```